bin/user/mqtt.py

```diff
@@ -390,7 +410,8 @@ def __init__(self, queue, server_url,
                  post_interval=None, stale=None,
                  log_success=True, log_failure=True,
                  timeout=60, max_tries=3, retry_wait=5,
-                 max_backlog=sys.maxsize):
+                 max_backlog=sys.maxsize,
+                 calculations={'dayRain':'day.rain.sum'}):
```

Comment:
Not sure whether the unconditional mentioning of rain is correct here?

Reply:
Yes, it is. See https://github.com/poblabs/weewx-belchertown/issues/685 for explanation.

Reply:
I am confused by this. Is this then a bug in the MQTT addon, which needs to be solved at the root rather than considered in your calculations contribution?? @matthewwall your opinion?

Reply:
I think so. The general rule of WeeWX is to have an interval, that is open at its left end and closed at its right end. The only exception is `dayRain`, and it is calculated that way because of the special needs of the the WU upload. Within the WeeWX sources there is a comment on this that is
```
                # NB: The WU considers the archive with time stamp 00:00
                # (midnight) as (wrongly) belonging to the current day
                # (instead of the previous day). But, it's their site,
                # so we'll do it their way.  That means the SELECT statement
                # is inclusive on both time ends:
```

For other use cases than the WU upload the daily rain should be calculated in the normal way, that is `$day.rain.sum`. 

The Belchertown skin uses `dayRain` to display the daily rain, and so it shows a wrong value when it rains just before midnight, if the MQTT extension uses that special calculated `dayRain` value.